hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java

```diff
@@ -234,7 +234,10 @@ private void loadFromSnapshotInfoTable(OMMetadataManager metadataManager) {
     // snapshot chains - both global and local path
     try (TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
              keyIter = metadataManager.getSnapshotInfoTable().iterator()) {
-      Map<Long, SnapshotInfo> snaps = new TreeMap<>();
+      Map<UUID, SnapshotInfo> snaps = new HashMap<>();
+      // Forward Linked list for snapshot chain.
+      Map<UUID, UUID> snapshotChain = new HashMap<>();
```

Comment:
How about using a LinkedList from java.util? We are using the map as linkedlist anyway.

Reply:
java.util LinkedList doesn't have setnext() ,setPrevious() methods. I guess it's fine to use map in that case.Ignore my comment.

Reply:
java.util.LinkedList search is bad. Since linkBefore function is package private. Only way to add is to use indexes which is bad. We will end up doing a complete iteration in the existing linked list to search for the node O(n^2). This solution when amortized O(n) as hashMap lookup is only O(1).